src/ui/App.js

```diff
@@ -11,18 +11,38 @@ import { useState, useCallback } from "react";
 export default function App({ initialCards }) {
   const [cards, setCards] = useState(initialCards);
   const [selected, setSelected] = useState([]);
+  const [includeAce, setIncludeAce] = useState(false);
+  const [isPlayAgain, setIsPlayAgain] = useState(false);
 
-  function toggleSelected(index) {
-    if (!selected.includes(index)) {
+function toggleSelected(index) {
+    if (!selected.includes(index) && selected.length < 3 ) {
       setSelected(selected.concat([index]));
-    } else {
+    }
+    else if (!selected.includes(index) && selected.length == 3){
+      if(cards[index].rank === 'A' && cards[index].suit === 'S'){
+	  // If there's an Ace, user can select - cards
+	  console.log('Ace is included');
+	  setSelected(selected.concat([index]));
+	  setIncludeAce(true);
+      }
+      else if(selected.some(index=>cards[index].rank === 'A' && cards[index].suit === 'S')){
+	  console.log('Ace is included');
+	  setSelected(selected.concat([index]));
+	  setIncludeAce(true);	  
+      }
+    }
+    else {
       setSelected(selected.filter((elt) => elt !== index));
     }
   }
 
   // This function will be called when the Draw button is clicked
   const fetchNewCards = useCallback(async () => {
-    console.log(`need to fetch ${selected.length} cards`);
+      console.log(`need to fetch ${selected.length} cards`);
+      if(includeAce == true){
```

Comment:
I don't understand this exactly what's happening here. Are you using the isPlayAgain state anywhere else?

It's probably fine for now, just figured I'd call it out.

Reply:
Ah, I see. So eventually you're going to add a playAgain button. That's a good idea.

However, that shouldn't have anything to do with having an Ace. It should automatically pop up after the draw phase. In other words, user gets dealt the initial hand, then chooses which cards to replace, then the new cards are dealt and the game is over. When the game is over, the Play Again button should be displayed.

Hope this makes sense!